mr/src/test/java/org/apache/iceberg/mr/TestCatalogs.java

```diff
@@ -256,6 +256,20 @@ public void testLoadCatalogHive() {
     Assertions.assertThat(hiveCatalog.get()).isInstanceOf(HiveCatalog.class);
   }
 
+  @Test
+  public void testLoadCustomCatalogWithHiveCatalogTypeSet() {
+    String catalogName = "barCatalog";
+    conf.set(InputFormatConfig.catalogPropertyConfigKey(catalogName, CatalogUtil.ICEBERG_CATALOG_TYPE),
+            CatalogUtil.ICEBERG_CATALOG_TYPE_HIVE);
+    conf.set(InputFormatConfig.catalogPropertyConfigKey(catalogName, CatalogProperties.CATALOG_IMPL),
+            CustomHadoopCatalog.class.getName());
+    conf.set(InputFormatConfig.catalogPropertyConfigKey(catalogName, CatalogProperties.WAREHOUSE_LOCATION),
+            "/tmp/mylocation");
+    Optional<Catalog> hiveCatalog = Catalogs.loadCatalog(conf, catalogName);
+    Assertions.assertThat(hiveCatalog.get()).isInstanceOf(CustomHadoopCatalog.class);
+    Assert.assertFalse("Catalog type shouldn't be Hive", Catalogs.hiveCatalog(conf, catalogName));
```

Comment:
personally I would prefer something like `Assertions.assertThat(Catalogs.getCatalogType(conf, catalogName)).isNotEqualTo(CatalogUtil.ICEBERG_CATALOG_TYPE_HIVE)` because it would tell you what the actual catalog type is when it fails, but that would require making `Catalogs.getCatalogType(..)` package-private

Reply:
Good point! I don't think there is a harm to change it to package-private and `@VisibleForTesting` annotation.